numba/cpython/unicode.py

```diff
@@ -593,6 +598,117 @@ def unicode_sub_check_type(ty, name):
         raise TypingError(msg)
 
 
+# FAST SEARCH algorithm implementation from cpython
+
+@register_jitable
+def _bloom_add(mask, ch):
+    mask |= (1 << (ch & (_BLOOM_WIDTH - 1)))
+    return mask
+
+
+@register_jitable
+def _bloom_check(mask, ch):
+    return mask & (1 << (ch & (_BLOOM_WIDTH - 1)))
+
+
+# https://github.com/python/cpython/blob/1960eb005e04b7ad8a91018088cfdb0646bc1ca0/Objects/stringlib/fastsearch.h#L550    # noqa: E501
+# XXX: Why we don't have the FAST_COUNT flag as cpython?
```

Comment:
CPython defines `FAST_COUNT` as 0. I think you can do the same as well.
https://github.com/python/cpython/blob/1960eb005e04b7ad8a91018088cfdb0646bc1ca0/Objects/stringlib/fastsearch.h#L24

Reply:
Actually, I think you implemented the `FAST_SEARCH` mode. So, the `mode` variable is not needed, and thus, unless you implement the three mode, the flags (`FAST_COUNT`, `FAST_SEARCH`, `FAST_RSEARCH`) defined in CPython will not be used.

Reply:
OK, I'll remove this question comment.